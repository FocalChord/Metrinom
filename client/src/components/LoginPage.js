import React from "react";

const LoginPage = () => {
    return (
        <div className="text-center">
            <header className="min-h-screen flex flex-col items-center justify-center bg-black text-white text-6xl">
                <a href="http://localhost:3001/auth/spotify" className="hover:text-green-500">
                    Login
                </a>
            </header>
        </div>
    );
};

export default LoginPage;
