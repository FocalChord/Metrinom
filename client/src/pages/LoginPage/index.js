import React from "react";
import { makeStyles, Container, Button, Paper } from "@material-ui/core";
import MetrinomLogo from "../../resources/MetrinomLogo.png";

const useStyles = makeStyles(() => ({
    root: {
        backgroundPosition: "center",
        backgroundRepeat: "no-repeat",
        backgroundSize: "cover",
    },
    containerItems: {
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
    },
    paperLayout: {
        width: "60%",
        margin: "10%",
        display: "flex",
        flexDirection: "column",
        textAlign: "center",
        alignItems: "center",
    },
    buttonLogin: {
        marginBottom: "5%",
        width: 350,
        height: 50,
        backgroundColor: "#1DB954",
        "&:hover": {
            backgroundColor: "#4AC776",
            color: "#FFF",
        },
    },
    textInput: {
        width: "75%",
    },
    title: {
        margin: "5%",
    },
}));

require("dotenv").config();

const METRINOME_BACKEND_URL = process.env.REACT_APP_URL || "http://localhost:3001";
const buttonLink = `${METRINOME_BACKEND_URL}/auth/spotify`;

const LoginPage = () => {
    const classes = useStyles();
    return (
        <div className={classes.root}>
            <Container className={classes.containerItems}>
                <Paper elevation={3} className={classes.paperLayout}>
                    <img alt="Metrinom Logo" style={{ width: 500 }} src={MetrinomLogo} />
                    <Button href={buttonLink} className={classes.buttonLogin} variant="contained" color="primary">
                        Sign in with Spotify
                    </Button>
                </Paper>
            </Container>
        </div>
    );
};
export default LoginPage;
