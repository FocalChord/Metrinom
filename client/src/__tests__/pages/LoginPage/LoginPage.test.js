import React from "react";
import renderer from "react-test-renderer";
import "regenerator-runtime/runtime.js";
import LoginPage from "../../../pages/LoginPage";

test("<RelatedArtistsGrid />", () => {
    const tree = renderer.create(<LoginPage />).toJSON();

    expect(tree).toMatchSnapshot();
});
