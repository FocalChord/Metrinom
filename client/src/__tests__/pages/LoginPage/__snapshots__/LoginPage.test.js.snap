// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RelatedArtistsGrid /> 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiContainer-root makeStyles-containerItems-2 MuiContainer-maxWidthLg"
  >
    <div
      className="MuiPaper-root makeStyles-paperLayout-3 MuiPaper-elevation3 MuiPaper-rounded"
    >
      <img
        alt="Metrinom Logo"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "width": 500,
          }
        }
      />
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-buttonLogin-4 MuiButton-containedPrimary"
        href="http://localhost:3001/auth/spotify"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          Sign in with Spotify
        </span>
      </a>
    </div>
  </div>
</div>
`;
